### DISCUSSION TITLE
Relationship of 'base' and 'master'.

### DISCUSSION THREAD

**Discussion by Texterity** [community/community#22323] [New to GitHub]
<p>This is a noob question (from techwriter with little Git experience using GitHub to build a documentation site.)</p>
<p>I have created a branch called ‘reviews’, which is the destination for the Edit Me button on the site pages. This workflow is intended to accomodate user reviews of any of the pages, so that site admin (me) can then review and merges changes into the default branch, which is master. As the site admin, I’m making changes directly to master.  (Incidentally I also intend to protect this branch so that other admins/users can’t merge/delete it.)</p>
<p>So far, so good.</p>
<p>This morning, I logged into GitHub and selected the review branch. Predictably, it says ‘This branch is 39 commits behind master.’ So I click on Pull Request to initiate a merge i.e. bring review up-to-date with master.</p>
<p>What is confusing to me is this window:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374.png" data-download-href="/uploads/short-url/pjcZ3eE5ADsTLcIL4jasK6Sa3iY.png?dl=1" title="Compare1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/38268019/181070179-9831f867-fb54-4aec-acca-f86ad5b48999.png" alt="Compare1.png" data-base62-sha1="pjcZ3eE5ADsTLcIL4jasK6Sa3iY" width="690" height="260" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374_2_690x260.png, https://aws1.discourse-cdn.com/github/original/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Compare1.png</span><span class="informations">822×310 31.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So here, the ‘base’ is set to ‘master’, and I want to compare ‘master’ with ‘review’.  But to do that, I find I need to set ‘base’ to be ’ <strong>review</strong>’, and ‘compare:’ to ’ <strong>master</strong>’. Then  I see the ‘Able to be merged’ and am able to enter a commit comment. </p>
<p>The reason I’m confused is the ambiguity of the relationship between ‘base’ and ‘master’.  I expected the ‘base’ to be equivalent to ‘master’ so that when a branch is being compared to it, it will display variances between ‘master’ and the other branch (in this case, ‘review’). So this seems counter-intuitive to me. </p>
<p>In any case, if I now go ahead and create a pull request, enter a comment (i.e. 'merging 39 changes for [date range]) then the result will be, the changes in master branch will be updated to review.</p>
<p>Is there anything in the documentation specifically about this issue? I have been looking around and there’s a lot that’s sort of relevant,  without making this exact point. </p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/texterity">@texterity</a>,</p>
<p>The Github UI says it right <strong>“This branch is 39 commits behind master”</strong> , there are changes on master and not in the review branch so there are nothing to merge to master.</p>
<p>Assume the following history:</p>
<p>                      A review<br>
                    /<br>
               A—B---C—D master</p>
<p>The branch <strong>review is created from the commit A of master</strong> while master continue to grow with some more commits B C D. In the review branch will not be add any more commits. Now you want to merge review into master, but the <strong>master had already the commit A so the review is </strong><strong>up-to-date with master</strong>. If you want the commits made on master you need to do the exact opposite: merge the master into review. In this case you put all the commits made on master into review. The example below become:</p>
<p>                 </p>
<p>                     A—B---C—D review<br>
                    /                /<br>
               A—B---C—D master</p>
<p>As a matter of fact, I try to do this in your <a href="https://github.com/Texterity/January" rel="nofollow noopener">repository</a> and this is the result:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8b6982e57f95d34ca439d69f5fa091cf78f98fd1.png" data-download-href="/uploads/short-url/jTiqn0DRlRJ7cTLIrHqa2Dj8og1.png?dl=1" title="testone1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070181-6a9ac83d-61cf-437f-b72c-33ea0cee7fd0.png" alt="testone1.png" data-base62-sha1="jTiqn0DRlRJ7cTLIrHqa2Dj8og1" width="565" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8b6982e57f95d34ca439d69f5fa091cf78f98fd1_2_565x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8b6982e57f95d34ca439d69f5fa091cf78f98fd1_2_847x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/8b6982e57f95d34ca439d69f5fa091cf78f98fd1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8b6982e57f95d34ca439d69f5fa091cf78f98fd1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">testone1.png</span><span class="informations">964×853 55.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<blockquote>
<p>What is confusing to me is this window:</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p></p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374.png" data-download-href="/uploads/short-url/pjcZ3eE5ADsTLcIL4jasK6Sa3iY.png?dl=1" title="Compare1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070184-dcbc0703-71e5-499b-9694-7bde7cb76710.png" alt="Compare1.png" data-base62-sha1="pjcZ3eE5ADsTLcIL4jasK6Sa3iY" width="690" height="260" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374_2_690x260.png, https://aws1.discourse-cdn.com/github/original/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/b1623cbfa8171801a07eb151d6b69c35ecdbb374_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Compare1.png</span><span class="informations">822×310 31.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</blockquote>
<blockquote>
<p>So here, the ‘base’ is set to ‘master’, and I want to compare ‘master’ with ‘review’.  But to do that, I find I need to set ‘base’ to be ’ <strong>review</strong>’, and ‘compare:’ to ’ <strong>master</strong>’. Then  I see the ‘Able to be merged’ and am able to enter a commit comment. </p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>The reason I’m confused is the ambiguity of the relationship between ‘base’ and ‘master’.  I expected the ‘base’ to be equivalent to ‘master’ so that when a branch is being compared to it, it will display variances between ‘master’ and the other branch (in this case, ‘review’). So this seems counter-intuitive to me.</p>
</blockquote>
<p>To explain more, the <strong>pull request is not a compare of file but a compare changes made by commits</strong>. If you imagine  <strong>the repository is not made by files but by commits</strong> is no more counter-intuitive.</p>
<blockquote>
<p>Is there anything in the documentation specifically about this issue? I have been looking around and there’s a lot that’s sort of relevant,  without making this exact point. </p>
</blockquote>
<p>You cannot find anything about this because there are no issues about that.</p>
<p>Hope this helps :slightly_smiling_face:  </p>
<p>-Gabriele-</p>
<blockquote>
<p><strong>Mark helpful posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great content!</strong></p>
</blockquote>

**Comment by Texterity** [comment]
<p>Thanks, Gabriele</p>
<blockquote>
<p>To explain more, the <strong>pull request is not a compare of file but a compare changes made by commits</strong>. If you imagine  <strong>the repository is not made by files but by commits</strong> is no more counter-intuitive.</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p> Hope this helps :slightly_smiling_face:  </p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>-Gabriele-</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<p>That is the concept that I was missing. Now that you have explained it so well, it makes perfect sense.</p>
